### PR TITLE
Remove `ConfigLoader` and `TemplatedConfigLoader`

### LINKED ISSUE #2692: Remove `ConfigLoader` and `TemplatedConfigLoader`
## Description
`OmegaConfigLoader` will be the config loader from `0.19.0` onwards.

## Context
https://github.com/kedro-org/kedro/milestone/34

## To do:
- [ ] Remove `ConfigLoader`
- [ ] Remove `TemplatedConfigLoader`
- [ ] Remove all other code related to these config loaders
- [ ] Remove any docs about these loaders and re-order the `OmegaConfigLoader` content where needed.
- [ ] Can we remove the `anyconfig` dependency? Double check if any other parts of our codebase are using it.

### PR DESCRIPTION
**Pull request by ankatiyar**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Resolve #2692 

## Development notes

- Remove `anyconfig` dependency
- Remove all code for `ConfigLoader` and `TemplatedConfigLoader`
- Update tests where needed
- Docs update in #3257 (docs PR needs to be merged first so the RTD builds pass)

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: kedro/config/__init__.py [hunk 1]
```diff
@@ -7,15 +7,15 @@
     BadConfigException,
     MissingConfigException,
 )
-from .config import ConfigLoader
+
+# from .config import ConfigLoader
 from .omegaconf_config import OmegaConfigLoader
-from .templated_config import TemplatedConfigLoader
+
+# from .templated_config import TemplatedConfigLoader
```
Comment (merelcht) [inline kedro/config/__init__.py:14]: Are these commented out ones still needed?

**Review by merelcht — APPROVED** [review]
Nice job on getting rid of `anyconfig` as well! ⭐ 👍

File: kedro/framework/cli/micropkg.py [hunk 2]
```diff
@@ -259,9 +260,9 @@ def _pull_package(  # noqa: too-many-arguments
 
 def _pull_packages_from_manifest(metadata: ProjectMetadata) -> None:
     # noqa: import-outside-toplevel
-    import anyconfig  # for performance reasons
+    import toml  # for performance reasons
```
Comment (astrojuanlu) [inline kedro/framework/cli/micropkg.py:263]: Is this still needed for performance reasons?

File: kedro/framework/cli/utils.py [hunk 3]
```diff
@@ -397,13 +397,13 @@ def _config_file_callback(ctx, param, value):  # noqa: unused-argument
     options are passed, they override config file values.
     """
     # for performance reasons
-    import anyconfig  # noqa: import-outside-toplevel
+    from omegaconf import OmegaConf
```
Comment (astrojuanlu) [inline kedro/framework/cli/utils.py:400]: Already being imported at the top of the file

File: pyproject.toml [hunk 4]
```diff
@@ -12,7 +12,6 @@ authors = [
 description = "Kedro helps you build production-ready data and analytics pipelines"
 requires-python = ">=3.8"
 dependencies = [
-    "anyconfig>=0.10.0",
```
Comment (astrojuanlu) [inline pyproject.toml:15]: 🎉

**Review by SajidAlamQB — APPROVED** [review]
Awesome, thanks @ankatiyar! 🚀

**Comment by astrojuanlu** [comment]
RTD failing with `sphinx.errors.SphinxWarning: /home/docs/checkouts/readthedocs.org/user_builds/kedro/checkouts/3253/docs/source/configuration/advanced_configuration.md:4:'myst' cross-reference target not found: '/kedro.config.ConfigLoader' [myst.xref_missing]`

**Comment by ankatiyar** [comment]
> RTD failing with `sphinx.errors.SphinxWarning: /home/docs/checkouts/readthedocs.org/user_builds/kedro/checkouts/3253/docs/source/configuration/advanced_configuration.md:4:'myst' cross-reference target not found: '/kedro.config.ConfigLoader' [myst.xref_missing]`

@astrojuanlu it's because of removing the code for the config loaders. The other PR #3257 will need to go in first!

**Pull request merged**